src/sys/statfs.rs

```diff
@@ -1,22 +1,244 @@
-use {Result, NixPath};
+use {NixPath, Result};
 use errno::Errno;
-use std::os::unix::io::AsRawFd;
 use libc;
+use std::mem;
+use std::os::unix::io::AsRawFd;
 
-pub fn statfs<P: ?Sized + NixPath>(path: &P, stat: &mut libc::statfs) -> Result<()> {
-    unsafe {
-        Errno::clear();
-        let res = try!(
-            path.with_nix_path(|path| libc::statfs(path.as_ptr(), stat))
-        );
+#[cfg(target_os = "android")]
+pub type fsid_t = libc::__fsid_t;
+#[cfg(not(target_os = "android"))]
+pub type fsid_t = libc::fsid_t;
+
+#[derive(Clone, Copy)]
+#[allow(missing_debug_implementations)]
+pub struct Statfs(libc::statfs);
+
+#[cfg(any(target_os = "ios", target_os = "macos", target_os="freebsd", target_arch = "s390x"))]
+#[derive(Eq, Copy, Clone, PartialEq, Debug)]
+pub struct FsType(libc::uint32_t);
+
+#[cfg(all(target_os = "linux", not(any(target_env = "musl", target_arch = "s390x"))))]
+#[derive(Eq, Copy, Clone, PartialEq, Debug)]
+pub struct FsType(libc::c_long);
+
+#[cfg(any(target_os = "android", all(target_os = "linux", target_env = "musl")))]
+#[derive(Eq, Copy, Clone, PartialEq, Debug)]
+pub struct FsType(libc::c_ulong);
+
+#[cfg(not(any(target_os = "linux", target_os = "ios", target_os = "macos", target_os = "android", target_os="freebsd")))]
```

Comment:
This `#[cfg()]` condition is getting very confusing.  Could you please convert it into the positive form?  Also wrap to 80 columns.